### PR TITLE
[SPARK-15416][SQL]Display a better message for not finding classes removed in Spark 2.0

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

If finding `NoClassDefFoundError` or `ClassNotFoundException`, check if the class name is removed in Spark 2.0. If so, the user must be using an incompatible library and we can provide a better message.
## How was this patch tested?
1. Run `bin/pyspark --packages com.databricks:spark-avro_2.10:2.0.1`
2. type `sqlContext.read.format("com.databricks.spark.avro").load("src/test/resources/episodes.avro")`.

It will show `java.lang.ClassNotFoundException: org.apache.spark.sql.sources.HadoopFsRelationProvider is removed in Spark 2.0. Please check if your library is compatible with Spark 2.0`

### CODE REVIEW

**Comment by zsxwing** [comment]
There are many space changes. Please use this link https://github.com/apache/spark/pull/13201/files?w=1 to review.

**Comment by zsxwing** [comment]
cc @marmbrus @yhuai

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala [hunk 1]
```diff
@@ -93,26 +101,45 @@ case class DataSource(
...
+                    error)
+                }
               }
+          }
+        } catch {
+          case e: NoClassDefFoundError => // This one won't be caught by Scala NonFatal
+            // NoClassDefFoundError's class name uses "/" rather than "." for packages
+            val className = e.getMessage.replaceAll("/", ".")
+            if (spark2RemovedClasses.contains(className)) {
+              throw new ClassNotFoundException(s"$className is removed in Spark 2.0. " +
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala]: nit `is -> was`

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala [hunk 2]
```diff
@@ -93,26 +101,45 @@ case class DataSource(
...
+                "Please check if your library is compatible with Spark 2.0")
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala]: Include `e` as the cause when throwing.

**Comment by marmbrus** [comment]
Thanks!  Merging to master and 2.0

**Pull request closed without merging**